scripts/ci/verify-dynamic-flavor-propagation.sh

```
@@ -0,0 +1,96 @@
+LOCAL_FLAVORS="build-logic/convention/src/main/kotlin/local/LocalFlavors.kt"
+MANIFEST="cmp-shared/build/kmp-flavors/variants.json"
+CANARY_FLAVOR="canary"
+CANARY_VARIANT="canaryRelease"
+BACKUP="$(mktemp)"
+
+log()  { printf '  [canary] %s\n' "$*"; }
+fail() { printf '  [canary] FAIL: %s\n' "$*" >&2; exit 1; }
+
+# ── guard: the derivation surfaces must be clean BEFORE we start, so the
+#    post-injection "untouched" assertion is meaningful. ─────────────────────
+DERIVATION_GLOBS=(deployment .github build-logic/convention/src/main/kotlin/local)
```

Shellcheck (0.11.0)

[warning] DERIVATION_GLOBS appears unused. Verify use (or export if used externally).

(SC2034)